guiEditor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -156,66 +156,73 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
         const adt = this.props.globalState.guiTexture;
         const content = JSON.stringify(adt.serializeContent());
 
-        const xmlHttp = new XMLHttpRequest();
-        xmlHttp.onreadystatechange = () => {
-            if (xmlHttp.readyState == 4) {
-                if (xmlHttp.status == 200) {
-                    const snippet = JSON.parse(xmlHttp.responseText);
-                    const oldId = adt.snippetId;
-                    adt.snippetId = snippet.id;
-                    if (snippet.version && snippet.version != "0") {
-                        adt.snippetId += "#" + snippet.version;
-                    }
-                    this.forceUpdate();
-                    if (navigator.clipboard) {
-                        navigator.clipboard.writeText(adt.snippetId);
-                    }
-
-                    const windowAsAny = window as any;
-
-                    if (windowAsAny.Playground && oldId) {
-                        windowAsAny.Playground.onRequestCodeChangeObservable.notifyObservers({
-                            regex: new RegExp(oldId, "g"),
-                            replace: `parseFromSnippetAsync("${adt.snippetId}`,
-                        });
-                    }
-
-                    this.props.globalState.customSave!.action(adt.snippetId).then(() => {
-                    }).catch((err) => {
+        if (this.props.globalState.customSave) {
+            this.props.globalState.customSave!.action(content).then((resolve) => {
+                adt.snippetId = (resolve as unknown as string);
+                this.forceUpdate();
+                if (navigator.clipboard) {
+                    navigator.clipboard.writeText(adt.snippetId);
+                }
+                alert("GUI saved with ID: " + adt.snippetId + " (please note that the id was also saved to your clipboard)");

```

Comment:
Do we want to use alert?

Reply:
Any reason why not?

Reply:
alert is blocking. But i guess that without some dialog mechanism there is no easy way of solving this. you can resolve this if there is no other way.